File: java/scf-simple/src/main/java/org/mydotey/scf/type/string/StringToIntConverter.java
package org.mydotey.scf.type.string;

/**
 * @author koqizhao
 *
 * May 21, 2018
 */
public class StringToIntConverter extends StringConverter<Integer> {

    public static final StringToIntConverter DEFAULT = new StringToIntConverter();

    public StringToIntConverter() {
        super(Integer.class);
    }

    @Override
    public Integer convert(String source) {
        return Integer.parseInt(source);
    }

}

File: dotnet/src/MyDotey.SCF.Simple/Type/String/StringToIntConverter.cs
using System;

namespace MyDotey.SCF.Type.String
{
    /**
     * @author koqizhao
     *
     * May 21, 2018
     */
    public class StringToIntConverter : StringConverter<int?>
    {
        public static readonly StringToIntConverter DEFAULT = new StringToIntConverter();

        public override int? Convert(string source)
        {
            if (string.IsNullOrWhiteSpace(source))
                return null;

            return int.Parse(source);
        }
    }
}